telegram/ext/filters.py

```diff
@@ -2185,6 +2202,37 @@ def filter(self, message: Message) -> bool:
 """This filter filters *any* message that has a :attr:`telegram.Message.from_user`."""
 
 
+class _UserAttachment(UpdateFilter):
+    __slots__ = ()
+
+    def filter(self, update: Update) -> bool:
+        return bool(update.effective_user) and bool(
+            update.effective_user.added_to_attachment_menu  # type: ignore
+        )
+
+
+USER_ATTACHMENT = _UserAttachment(name="filters.USER_ATTACHMENT")
+"""This filter filters *any* message that have a user who added the bot to their
+:attr:`attachment menu <telegram.User.added_to_attachment_menu>` as
+:attr:`telegram.Update.effective_user`.
+"""
+
+
+class _UserPremium(UpdateFilter):
+    __slots__ = ()
+
+    def filter(self, update: Update) -> bool:
+        return bool(update.effective_user) and bool(
+            update.effective_user.is_premium  # type: ignore
+        )
+
+
+USER_PREMIUM = _UserPremium(name="filters.USER_PREMIUM")
```

Comment:
```suggestion
PREMIUM_USER= _UserPremium(name="filters.USER_PREMIUM")
```
`USER_PREMIUM` sounds odd